sdk/src/client/secret/ledger_nano.rs

```diff
@@ -172,8 +172,55 @@ impl SecretManage for LedgerSecretManager {
         Err(Error::UnsupportedOperation.into())
     }
 
-    async fn sign_ed25519(&self, _msg: &[u8], _chain: &Chain) -> Result<Ed25519Signature, Self::Error> {
-        Err(Error::UnsupportedOperation.into())
+    async fn sign_ed25519(&self, msg: &[u8], chain: &Chain) -> Result<Ed25519Signature, Self::Error> {
+        if msg.len() != 32 {
+            return Err(Error::UnsupportedOperation.into());
```

Comment:
Other error would be good or at least have a comment that it's limited to 32 bytes

Reply:
Tbh I think this ones fits, signing anything else than 32 bytes is an unsupported operation. But yeah I'll add a comment.

Reply:
Added